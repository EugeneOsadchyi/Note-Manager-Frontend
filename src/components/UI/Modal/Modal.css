.Modal {
  position: fixed;
  top: 30%;
  left: calc(50% - 250px);
  width: 500px;
  padding: 16px;
  border: 2px solid black;
  background-color: white;
  box-shadow: 1px 1px 1px black;
  text-align: center;
  z-index: 500;
}
