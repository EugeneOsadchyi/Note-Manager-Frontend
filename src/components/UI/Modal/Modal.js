import React from 'react';
import PropTypes from 'prop-types';

import './Modal.css';

import Aux from '../../../hoc/Aux';
import Backdrop from '../Backdrop';

const Modal = ({ children, show, modalClosed }) => (
  <Aux>
    <Backdrop show={show} clicked={modalClosed} />
    <div className="Modal">
      {children}
    </div>
  </Aux>
);

Modal.propTypes = {
  children: PropTypes.element.isRequired,
  show: PropTypes.bool.isRequired,
  modalClosed: PropTypes.func.isRequired,
};

export default Modal;
