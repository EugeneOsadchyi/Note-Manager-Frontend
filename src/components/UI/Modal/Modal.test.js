import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Modal from './Modal';

describe('Modal', () => {
  const mockModalClosed = jest.fn();

  const children = (
    <h1>
      Modal content
    </h1>
  );

  const props = {
    show: true,
    modalClosed: mockModalClosed,
  };

  const modal = shallow((
    <Modal {...props}>
      {children}
    </Modal>
  ));

  it('renders properly', () => {
    expect(shallowToJson(modal)).toMatchSnapshot();
  });

  it('contains Modal element', () => {
    expect(modal.find('.Modal').exists()).toBe(true);
  });

  it('contains Backdrop component', () => {
    expect(modal.find('Backdrop').exists()).toBe(true);
  });

  describe('Backdrop component', () => {
    it('receives `show` and `clicked` props', () => {
      const { show, modalClosed } = props;

      const expectedProps = {
        show,
        clicked: modalClosed,
      };

      expect(modal.find('Backdrop').props()).toEqual(expectedProps);
    });
  });
});
