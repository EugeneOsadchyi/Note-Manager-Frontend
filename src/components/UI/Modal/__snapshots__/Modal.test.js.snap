// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders properly 1`] = `
<Component>
  <Backdrop
    clicked={[Function]}
    show={true}
  />
  <div
    className="Modal"
  >
    <h1>
      Modal content
    </h1>
  </div>
</Component>
`;
